src/utils/viewer/viewer.cpp

```diff
@@ -359,7 +360,11 @@ void Viewer::drawEvent() {
 void Viewer::viewportEvent(ViewportEvent& event) {
   GL::defaultFramebuffer.setViewport({{}, framebufferSize()});
   renderCamera_->getMagnumCamera().setViewport(event.windowSize());
```

Comment:
The `SceneGraph::Camera` class can handle aspect ratio preservation in the `setViewport()` call automatically -- you only need to call [Camera::setAspectRatioPolicy()](https://doc.magnum.graphics/magnum/classMagnum_1_1SceneGraph_1_1Camera.html#a98f200ca05c1e0a7fa06845b3510f8f2) with a desired value.

Looking at the code, [here](https://github.com/facebookresearch/habitat-sim/blob/eb191d19165a32101a3745bde03518ddfe86155e/src/esp/gfx/RenderCamera.cpp#L36) it's set to `NotPreserved` for some reason, which effectively disables such handling.

Reply:
I see. Good to know. I do not know the purpose either. Will mark it, and solve it in recent "sensor/camera redesign" project (See slack for more details. Hope you can participate in this project. :-) )

Reply:
Well, I commented here because doing it via `setAspectRatioPolicy()` would result in *less code* overall. Merging this as-is just piled up more unnecessary tech debt in my opinion.

`NotPreserved` is the default, and it probably makes sense for rendering headless -- there you have no arbitrary framebuffer resizes anyway. But having that there doesn't prevent the *viewer* from redefining it to handle this differently when having a GUI.

Reply:
Completely agree with @mosra .  There is no reason to keep the `NotPreserved` here and add extra code to implement behavior we could achieve by just changing the flag.  Let's have a followup PR to remove the extra code + extra comments, and just switch the flag instead.

Reply:
I was thinking when @msavva wrote it, he must have some reasons, so I was trying to respect that by not touching it, and leaving it to the future discussion.
But since he also leans towards changing it now, I am happy to write a new PR to roll back this one, and fix the issue in the other way.